new_platforms/docs/new_platform_dev.md

```diff
@@ -1,392 +1,462 @@
-"Configuration Properties"->"C/C++"->"Preprocessor" properties for All Platforms
-for the DebugOpteeSimulation configuration, add
-**OE\_SIMULATE\_OPTEE**.  Then do the same for your enclave project.
-3. Add oehost\_opteesim.lib to the Additional Dependencies of your app and
-remove any sgx libraries, for All Platforms for the DebugOpteeSimulation
-configuration. Your libs might look like this:
-"oehost.lib;ws2\_32.lib;rpcrt4.lib;shell32.lib;oehost\_opteesim.lib"
-4. Your app Additional Include Directories for DebugOpteeSimulation
-should include at least:
-* $(NewPlatformsDir)include\optee\host
-* $(NewPlatformsDir)include\optee
-* $(NewPlatformsDir)include
-* $(SGXSDKInstallPath)\include
-5. In the "Configuration Properties->"Debugging", change Debugger to launch
-back to Local Windows Debugger, and make the working directory be the
-directory your enclave is built in (usually "$(OutDir)").
-6. The same files that you build for OP-TEE should be built for
-DebugOpteeSimulation, so if you have files selectively built (e.g., if you
-have C files marked as Exclude From Build for certain configurations),
-update your configuration so that the same files get built, not any
-SGX-specific ones.
-7. In your TA's "Configuration Properties"->"C/C++"->"General" properties of the
-DebugOpteeSimulation configuration for All Platforms, the Additional Include
-Directories should NOT include $(SGXSDKInstallPath)include\tlibc or
-$(SGXSDKInstallPath)include\libc++, and should include at least:
-* $(OESdkDir)3rdparty\RIoT\CyReP\cyrep
-* $(NewPlatformsDir)include\optee\enclave\Simulator
-* $(NewPlatformsDir)include\optee\enclave
-* $(NewPlatformsDir)include\optee
-* $(NewPlatformsDir)include
-* $(SGXSDKInstallPath)include
-* $(OESdkDir)3rdparty\optee\_os\lib\libutee\include
-* $(OESdkDir)3rdparty\optee\_os\lib\libutils\ext\include
-8. In your TA's "Configuration Properties"->"VC++ Directories" properties of the
-DebugOpteeSimulation configuration for All Platforms, change the Include Directories
-and Library Directories from $(NoInherit) back to \<inherit from parent or
-project defaults\>.
-9. In your TA's "Configuration Properties"->"Linker"->"Input" in the
-DebugOpteeSimulation configuration for All Platforms, clear "Ignore All Default
-Libraries", and add oeenclave\_opteesim.lib to the Additional Dependencies of
-your TA, and remove any sgx libs.  Your libs might look like this:
-"oeenclave.lib;oeenclave\_opteesim.lib;kernel32.lib;vcruntime.lib;ucrtd.lib"
-10. In your TA's "Configuration Properties"->"Build Events"->"Post-Build Event",
-change the "Use in Build" to No for the DebugOpteeSimulation configuration for all
-Platforms.
-11. In your TA's "Configuration Properties"->"General" section in the
-DebugOpteeSimulation configuration for All Platforms, change the Target Name
-to be your TA's UUID (without curly braces).
 
-# Understanding the SDK
+# Using the SDK
+
+The Open Enclave SDK uses [CMake](https://cmake.org) as its build system for
```

Comment:
True but this should be irrelevant for this page.  This page is about what the app uses, not what the OE SDK uses.   The app should not need to use cmake, especially on Windows.

Reply:
True. There are two ways to use the SDK as far as I can tell: one, via the template you are working on, and two, by pulling the SDK as a submodule. Since the template isn't public yet, I didn't mention it, so that leaves the second option. For the second option, there has to be some way to import the SDK into a new project, but that can't be a regular VS solution because those don't integrate with CMake.